android/expoview/src/main/java/host/exp/expoview/Exponent.java

```diff
@@ -480,6 +483,26 @@ public void onCachedResponse(ExpoResponse response, boolean isEmbedded) {
     return sourceFile.exists();
   }
 
+  // As OTAs piles up, the JS bundles will consume quite an amount of storage space
+  // App developers, if find needed, can purge all the existing cache
+  public void clearAllJSBundleCache(final String abiVersion) {
+    final File filesDir = mContext.getFilesDir();
+    final File directory = new File(filesDir, abiVersion);
+    if (!ABIVERSION_PATTERN.matcher(abiVersion).matches()) {
+      return;
+    }
+    try {
+      if (!directory.getCanonicalPath().startsWith(filesDir.getCanonicalPath())) {
```

Comment:
what is this check needed for? Based on how you're creating the `directory` variable it seems like this condition would never be false.

Reply:
It can be false, if the developer injects an `abiVersion` to be a combination of `..` and other path components. Without this check, developers can traverse the filesystem arbitrarily.

Reply:
ah, good catch. 👍